sdk/core/Azure.Core/tests/TestFramework/TestRecording.cs

```diff
@@ -0,0 +1,149 @@
+            return RecordSession.Deserialize(jsonDocument.RootElement);
+        }
+
+        public void Start()
+        {
+            switch (Mode)
+            {
+                case RecordedTestMode.Record:
+                    _session = new RecordSession();
+                    break;
+                case RecordedTestMode.Playback:
+                    _session = Load();
+                    break;
+            }
+        }
+
+        public void Stop()
+        {
+            if (Mode == RecordedTestMode.Record)
+            {
+                var directory = Path.GetDirectoryName(_sessionFile);
+                Directory.CreateDirectory(directory);
+
+                using FileStream fileStream = File.Create(_sessionFile);
+                var utf8JsonWriter = new Utf8JsonWriter(fileStream, new JsonWriterOptions()
+                {
+                    Indented = true
+                });
+                _session.Sanitize(_sanitizer);
+                _session.Serialize(utf8JsonWriter);
+                utf8JsonWriter.Flush();
+            }
+        }
+        public HttpPipelineTransport CreateTransport(HttpPipelineTransport currentTransport)
+        {
+            switch (Mode)
+            {
+                case RecordedTestMode.None:
+                    return currentTransport;
+                case RecordedTestMode.Record:
+                    return new RecordTransport(_session, currentTransport);
+                case RecordedTestMode.Playback:
+                    return new PlaybackTransport(_session, _matcher);
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(Mode), Mode, null);
+            }
+        }
+
+        public void SetConnectionString(string name, string connectionString)
+        {
+            ConnectionString s = ConnectionString.Parse(connectionString);
+            _sanitizer.SanitizeConnectionString(s);
+            _session.Variables[name] = s.ToString();
+        }
+
+        public string GetConnectionString(string name)
+        {
+            return _session.Variables[name];
+        }
+
+        public string GenerateId()
+        {
+            return Random.Next().ToString();
+        }
+
+        public string GetConnectionStringFromEnvironment(string variableName)
+        {
+            var environmentVariableValue = Environment.GetEnvironmentVariable(variableName);
+            switch (Mode)
+            {
+                case RecordedTestMode.Record:
+                    SetConnectionString(variableName, environmentVariableValue);
+                    return environmentVariableValue;
+                case RecordedTestMode.None:
+                    return environmentVariableValue;
+                case RecordedTestMode.Playback:
+                    return GetConnectionString(variableName);
+                default:
+                    throw new ArgumentOutOfRangeException();
+            }
+        }
```

Comment:
I think this method is too specific.  Shouldn't we have the ability to do this with any environment variable?

Reply:
Done